core-container/entry.sh

```diff
@@ -23,145 +23,133 @@ function echoerr() {
    echo "ERROR: $1" >&2
 }
 
-
-function set_provider_adapters() {
-    oldKubeconfig=$KUBECONFIG
-    KUBECONFIG=
-    echo "Waiting for space manager to be ready... this may take a while."
+function wait_kcp_ready() {
+    echo "Waiting for kcp to be ready... this may take a while."
     (
-        until [ "$(kubectl logs $(kubectl get pod --selector=app=kubestellar -o jsonpath='{.items[0].metadata.name}') -c space-manager | grep '***READY***')" != "" ]; do
-            sleep 10
+        until [ "$(kubectl --kubeconfig $host_kubeconfig logs $(kubectl --kubeconfig $host_kubeconfig get pod --selector=app=kubestellar -o jsonpath='{.items[0].metadata.name}') -c kcp | grep '***READY***')" != "" ]; do
+           sleep 10
         done
     )
+}
+
+function get_kcp_kubeconfig() {
+    wait_kcp_ready
+    kcp_kubeconfig_dir=/home/kubestellar/.kcp
+    kcp_kubeconfig=${kcp_kubeconfig_dir}/admin.kubeconfig
+    echo "Copying the admin.kubeconfig from kubestellar seret..."
+    mkdir -p $kcp_kubeconfig_dir
+    (
+        kubectl --kubeconfig $host_kubeconfig get secrets kubestellar -o 'go-template={{index .data "admin.kubeconfig"}}' | base64 --decode > $kcp_kubeconfig
+    )
+}
 
+function create_kcp_provider() {
+    get_kcp_kubeconfig     # kcp is created in a seperate container 
     echo "Delete the kcp provider secret if it already exists, and then create it."
-    if ! kubectl delete secret -n ${NAMESPACE} kcpsec ; then
+    if ! kubectl --kubeconfig $SPACE_MANAGER_KUBECONFIG delete secret -n ${NAMESPACE} kcpsec ; then
         echo "Nothing to delete."
     fi
-    kubectl get secrets kubestellar -o 'go-template={{index .data "admin.kubeconfig"}}' | base64 --decode > kcpsecret
-    kubectl create secret generic -n ${NAMESPACE} kcpsec --from-file=kubeconfig="kcpsecret"    
+    kubectl --kubeconfig $host_kubeconfig get secrets kubestellar -o 'go-template={{index .data "admin.kubeconfig"}}' | base64 --decode > kcpsecret
```

Comment:
This looks like it is duplicating functionality from `get_kcp_kubeconfig`. Why have that duplication?